hydra-node/src/Hydra/Options.hs

```diff
@@ -534,6 +534,38 @@ hydraNodeCommand =
       (decodeUtf8 $ encodePretty Contract.scriptInfo)
       (long "script-info" <> help "Dump script info as JSON")
 
+data InvalidOptions
+  = MaximumNumberOfPartiesExceeded
+  | CardanoAndHydraKeysMissmatch
+  deriving (Eq, Show)
+
+-- | Hardcoded limit for maximum number of parties in a head protocol
+-- NB: we don't count our own node here so need to do + 1 at the call site
+maximumNumberOfParties :: Int
+maximumNumberOfParties = 4
+
+explain :: InvalidOptions -> String
+explain = \case
+  MaximumNumberOfPartiesExceeded -> "Maximum number of parties is currently set to: " <> show maximumNumberOfParties
+  CardanoAndHydraKeysMissmatch -> "Number of loaded cardano and hydra keys needs to match"
+
+-- | Validate cmd line arguments for hydra-node and check if they make sense before actually running the node.
+-- Rules we apply:
+--  - Check if number of parties is bigger than our hardcoded limit
+--      (by looking at loaded hydra or cardano keys and comparing it to the 'maximumNumberOfParties')
+--  - Check that number of loaded hydra keys match with the number of loaded cardano keys
+--      (by comparing lengths of the two lists)
+validateRunOptions :: RunOptions -> Either InvalidOptions ()
+validateRunOptions RunOptions{hydraVerificationKeys, chainConfig}
+  | numberOfParties > maximumNumberOfParties + 1 = Left MaximumNumberOfPartiesExceeded
```

Comment:
```suggestion
  | numberOfParties +1 > maximumNumberOfParties = Left MaximumNumberOfPartiesExceeded
```

It's plus one, but not because the maximum excludes us. It's because the number of Parties is excluding us. Feel free to add a comment here or name it `numberOfOtherParties` or so.